server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -355,14 +381,29 @@ synchronized void cleanCache() {
                 }
             }
         }
+        refresh(CacheStoreType.ON_HEAP);
+    }
+
+    private Iterable<Key> getIterable(CacheStoreType cacheStoreType) {
+        if (cache instanceof TieredCache) {
+            return ((TieredCache<Key, BytesReference>) cache).cacheKeys(cacheStoreType);
+        } else {
+            return cache.keys();
+        }
+    }
 
-        cache.refresh();
+    private void refresh(CacheStoreType cacheStoreType) {
+        if (cache instanceof TieredCache) {
+            ((TieredCache<Key, BytesReference>) cache).refresh(cacheStoreType);
+        } else {
+            cache.refresh();
+        }
```

Comment:
Why we need these separate logic for `instanceOf` ? Any `Cache` implementation should support the `keys()` and `refresh()` mechanism. For `TieredCache` if it has both the `onHeap` and `onDisk` it will take care of handling for both the caches otherwise if only `onHeap` is present then it will perform these operation on that cache only.

Reply:
Here I wanted a capability to return a key iterator or call refresh for a certain storeType for TieredCache use case. 
Otherwise doing a keys() or refresh() as a whole for TieredCache will trigger for both onHeap/disk. 

Couldn't think of a better way at this point.

Reply:
We are looking to do clean up of specific types of cache on separate threads so that we can control the frequency of each the cleanup background job. This is necessary as we may want to maybe do clean up for disk with a longer duration compared to onheap cache.
With this use case it necessary to iterator over keys separately.

Reply:
Can that `cleanup` logic be kept inside the TieredCache then ? Because cleaning for specific store type is only valid for TieredCache. For other concrete caches it will cleanup all their keys.

Reply:
It is tricky considering IndicesRequestCache has its own internal logic to clean up. Anyway I will raise another PR to introduce a top level setting to choose cache type and then accordingly integrate TieredCache, we can discuss this there.